nvflare/app_opt/xgboost/histogram_based_v2/runners/xgb_client_runner.py

```diff
@@ -109,40 +115,74 @@ def run(self, ctx: dict):
         self._rank = ctx.get(Constant.RUNNER_CTX_RANK)
         self._world_size = ctx.get(Constant.RUNNER_CTX_WORLD_SIZE)
         self._num_rounds = ctx.get(Constant.RUNNER_CTX_NUM_ROUNDS)
+        self._training_mode = ctx.get(Constant.RUNNER_CTX_TRAINING_MODE)
+        self._xgb_params = ctx.get(Constant.RUNNER_CTX_XGB_PARAMS)
+        self._xgb_options = ctx.get(Constant.RUNNER_CTX_XGB_OPTIONS)
         self._server_addr = ctx.get(Constant.RUNNER_CTX_SERVER_ADDR)
         # self._data_loader = ctx.get(Constant.RUNNER_CTX_DATA_LOADER)
         self._tb_dir = ctx.get(Constant.RUNNER_CTX_TB_DIR)
         self._model_dir = ctx.get(Constant.RUNNER_CTX_MODEL_DIR)
 
-        if self.use_gpus:
+        use_gpus = self._xgb_options.get("use_gpus", False)
+        if use_gpus:
             # mapping each rank to a GPU (can set to cuda:0 if simulating with only one gpu)
             self.logger.info(f"Training with GPU {self._rank}")
-            self.xgb_params["device"] = f"cuda:{self._rank}"
-
-        self.logger.info(f"Using xgb params: {self.xgb_params}")
-        params = XGBoostParams(
-            xgb_params=self.xgb_params,
-            num_rounds=self._num_rounds,
-            early_stopping_rounds=self.early_stopping_rounds,
-            verbose_eval=self.verbose_eval,
-        )
+            self._xgb_params["device"] = f"cuda:{self._rank}"
 
+        self.logger.info(
+            f"XGB trainging_mode: {self._training_mode} " f"params: {self._xgb_params} XGB options: {self._xgb_options}"
+        )
         self.logger.info(f"server address is {self._server_addr}")
+
         communicator_env = {
-            "dmlc_communicator": "federated",
+            "xgboost_communicator": "federated",
             "federated_server_address": f"{self._server_addr}",
             "federated_world_size": self._world_size,
             "federated_rank": self._rank,
-            # FIXME: It should be possible to customize this or find a better location
-            # to distribut the shared object, preferably along side the nvflare Python
-            # package.
-            "federated_plugin": {"path": "/tmp/libproc_nvflare.so"},
         }
+
```

Comment:
what is the correct way to work with XGBoost's latest code of secure xgboost?

Reply:
This is still coded against current API. Need another PR to work with XGBoost 2.2 API.